Not respecting `context` in webpack.config.js

<!--
  Issues are so 🔥

  If you remove or skip this template, you'll make the 🐼 sad and @ken_wheeler
  will appear and pile-drive the close button from a great height while making
  animal noises.

  👉🏽 Need support, advice, or help? Don't open an issue.
  Head to StackOverflow or https://gitter.im/webpack/webpack.
-->

* Operating System: macOS 10.13.4
* Node Version: v10.4.1
* Yarn Version: 1.6.0
* webpack Version: 4.12.0
* webpack-command Version: 0.2.1

<!-- Please place an x (no spaces!) in all [ ] that apply -->

This issue is for a:

- [x] **bug**
- [ ] **feature** request
- [ ] **modification** request

### Code
https://github.com/alexilyaev/react-es6-starter/tree/webpack-command-context-bug

##### CLI Command

```console
  $ npm run build
```

##### webpack.config.js

https://github.com/alexilyaev/react-es6-starter/blob/webpack-command-context-bug/webpack.config.js

### Expected Behavior

`npm run build` should succeed.

### Actual Behavior

Fails with:

```
> cross-env NODE_ENV=production webpack

clean-webpack-plugin: /Users/alex/www/Node/Personal/react-es6-starter/dist has been removed.
ℹ ｢webpack｣: Starting Build
ℹ ｢webpack｣: Build Finished

webpack v4.12.0

475330bd3c812c147ff4
  size  name  module  status

  Δt 581ms


Entry module not found: Error: Can't resolve './app.js' in '/Users/alex/www/Node/Personal/react-es6-starter'
  0:0  error  webpack-stylish: <please report unknown message format>

Error: Child compilation failed:
  253:16  error  Entry module not found: Error: Can't resolve
                 '/Users/alex/www/Node/Personal/react-es6-starter/app.html' in
                 '/Users/alex/www/Node/Personal/rea  ct-es6-starter':   Error: Can't
                 resolve '/Users/alex/www/Node/Personal/react-es6-starter/app.html' in
                 '/Users/alex/www/Node/Personal/react-es6-starter'      - compiler.js:79
                 childCompiler.runAsChild
                 [react-es6-starter]/[html-webpack-plugin]/lib/compiler.js:      -
                 Compiler.js:243 compile
                 [react-es6-starter]/[webpack]/lib/Compiler.js:243:11      -
                 Compiler.js:500 hooks.afterCompile.callAsync.err
                 [react-es6-starter]/[webpack]/lib/Compiler.js:500:14         -
                 Hook.js:35 AsyncSeriesHook.lazyCompileHook [as _callAsync]
                 [react-es6-starter]/[tapable]/lib/Hook.js:35:21      - Compiler.js:497
                 compilation.seal.err
                 [react-es6-starter]/[webpack]/lib/Compiler.js:497:30         -
                 Hook.js:35 AsyncSeriesHook.lazyCompileHook [as _callAsync]
                 [react-es6-starter]/[tapable]/lib/Hook.js:35:21      -
                 Compilation.js:986 hooks.optimizeAssets.callAsync.err
                 [react-es6-starter]/[webpack]/lib/Compilation.js:986:35         -
                 Hook.js:35 AsyncSeriesHook.lazyCompileHook [as _callAsync]
                 [react-es6-starter]/[tapable]/lib/Hook.js:35:21      -
                 Compilation.js:977 hooks.optimizeChunkAssets.callAsync.err
                 [react-es6-starter]/[webpack]/lib/Compilation.js:977:32         -
                 index.js:334
                 [react-es6-starter]/[uglifyjs-webpack-plugin]/dist/index.js:334:11
                 - index.js:63 _class.runTasks
                 [react-es6-starter]/[uglifyjs-webpack-plugin]/dist/uglify/index.js:63:9
                 - index.js:253 UglifyJsPlugin.optimizeFn

✖ 2 problems (2 errors, 0 warnings)
```

### How Do We Reproduce?

- Clone
- `yarn`
- `npm run build`

## Other Notes

- `context` is not respected
- Works ok with webpack-cli

I've been hitting this bug as well with v0.2.1 (it works fine on v0.2.0).

* Operating System: macOS 10.13.5
* Node Version: 10.3.0
* NPM Version: 6.1.0
* webpack Version: 4.12.0
* webpack-command Version: 0.2.1

<!-- Please place an x (no spaces!) in all [ ] that apply -->

This issue is for a:

- [x] **bug**
- [ ] **feature** request
- [ ] **modification** request

### Code

##### CLI Command

```console
  $ webpack
```

##### webpack.config.js

```js
const path = require('path');
const webpack = require('webpack');

module.exports = {
  mode: 'development',
  // context for entry points
  context: path.join(__dirname, 'static', 'js'),
  // define all the entry point bundles
  entry: {
    app: './init.js',
  },
  output: {
    path: path.join(__dirname, 'content', 'static', 'assets'),
    publicPath: '/static/assets/',
    filename: '[name].bundle.js',
  },
  resolve: {
    // where to look for "required" modules
    modules: ['static/js', 'templates', 'sass', 'static', 'node_modules'],
    alias: { jquery: 'jquery/dist/jquery.slim.js' },
  },
  optimization: {
    runtimeChunk: 'single',
    splitChunks: {
      name: false,
    },
  },
  plugins: [
    // make jquery accessible in all modules that use it
    new webpack.ProvidePlugin({
      $: 'jquery',
      jQuery: 'jquery',
      'window.jQuery': 'jquery',
      'root.jQuery': 'jquery',
    }),
  ],
  module: {
    rules: [
      {
        test: /(static\/js\/.*\.js$|test\/.*\.js$)/,
        exclude: /(node_modules|vendor)/,
        use: [
          {
            loader: 'babel-loader',
            options: { cacheDirectory: true },
          },
        ],
      },
    ],
  },
};
```

### Expected Behavior

`webpack-command` v0.2.0:

```
ℹ ｢webpack｣: Starting Build
ℹ ｢webpack｣: Build Finished

webpack v4.12.0

d8770d5d615893b45708
  size     name     module             status
  5.38 kB  base.js  ./app/base.js      built
  152 B    init.js  ./init.js          built

  size     name     asset              status
  301 kB   app      app.bundle.js      emitted
  6.2 kB   runtime  runtime.bundle.js  emitted

  Δt 1425ms (35 modules hidden)
```

### Actual Behavior

`webpack-command` v0.2.1:

```
ℹ ｢webpack｣: Starting Build
ℹ ｢webpack｣: Build Finished

webpack v4.12.0

a293ba96e559b9f271af
  size  name  module  status

  Δt 106ms


Entry module not found: Error: Can't resolve './init.js' in '/Users/jonny/Sites/oddsite'
  0:0  error  webpack-stylish: <please report unknown message format>

✖ 1 problem (1 error, 0 warnings)
```

### How Do We Reproduce?

From repo: https://github.com/oddbird/oddsite/tree/webpack-command-bug
Run `npm install` (on `webpack-command-bug` branch), and then `webpack` (or `./node_modules/.bin/webpack`).

--- a/lib/config.js
+++ b/lib/config.js
@@ -7,12 +7,13 @@ const WebpackCommandError = require('./WebpackCommandError');
 
 module.exports = {
   distill(argv, config, options) {
+    const cwdContext = { context: process.cwd() };
     let result;
 
     if (Array.isArray(config)) {
-      result = config.map((conf) => merge(conf, options));
+      result = config.map((conf) => merge(cwdContext, conf, options));
     } else {
-      result = merge(config, options);
+      result = merge(cwdContext, config, options);
     }
 
     if (argv.configName) {

--- a/lib/flags/general.js
+++ b/lib/flags/general.js
@@ -40,9 +40,9 @@ module.exports = {
 
     if (argv.context) {
       result.context = path.resolve(argv.context);
-    } else {
-      result.context = process.cwd();
     }
+    // there is no else case for context, as we don't want a default value
+    // here to override a context value in a config
 
     if (argv.debug) {
       const { LoaderOptionsPlugin } = webpack;

--- a/test/tests/config.js
+++ b/test/tests/config.js
@@ -1,5 +1,6 @@
 const { test } = require('../util');
 const { distill } = require('../../lib/config');
+const { apply } = require('../../lib/flags');
 
 const config = [
   {
@@ -16,6 +17,20 @@ test('lib/config', module, () => {
   it(`distill()`, () => {
     const result = distill({}, config, {});
     expect(result).toMatchSnapshot();
+  });
+
+  it(`distill() context`, () => {
+    const argv = {};
+    const options = apply({}, {});
+
+    let result = distill(argv, { context: 'batman' }, { context: 'superman' });
+    expect(result.context).toBe('superman');
+
+    result = distill(argv, { context: 'batman' }, options);
+    expect(result.context).toBe('batman');
+
+    result = distill(argv, {}, options);
+    expect(result.context).toBe(process.cwd());
   });
 
   it(`distill() plugins`, () => {

